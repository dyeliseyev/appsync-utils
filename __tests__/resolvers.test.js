/* test full resolver pipelines */

import { checkResolverValid } from "./helpers";
import { util } from "..";

describe("dynamodb resolvers", () => {
  test("something", async () => {
    const code = `
    export function request(ctx) {
        return {
            operation: "Query",
            index: "nameIndex",
            select : "ALL_ATTRIBUTES",
            query: {
                "expression" : "#name = :name",
                "expressionNames": {
                    "#name": "name",
                },
                "expressionValues" : {
                    ":name" : util.dynamodb.toDynamoDB(ctx.arguments.filter.line),
                }
            },
            filter: {
                "expression" : "#shift = :shift",
                "expressionNames": {
                    "#shift": "shift",
                },
                "expressionValues" : {
                    ":shift" : util.dynamodb.toDynamoDB(ctx.arguments.filter.shift),
                }
            },
        };
    }

    export function response(ctx) {
        return ctx.result.items;
    }
    `;

    const requestContext = {
      arguments: {
        filter: {
          line: "test",
          shift: 10,
        },
      },
    };

    await checkResolverValid(code, requestContext, "request");

    const responseContext = {
      result: {
        items: [
          { a: 10 },
        ],
      },
    };

    await checkResolverValid(code, responseContext, "response");
  });
});
