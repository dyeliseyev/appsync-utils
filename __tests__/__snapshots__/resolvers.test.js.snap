// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamodb resolvers something 1`] = `
{
  "filter": {
    "expression": "#shift = :shift",
    "expressionNames": {
      "#shift": "shift",
    },
    "expressionValues": {
      ":shift": {
        "N": 10,
      },
    },
  },
  "index": "nameIndex",
  "operation": "Query",
  "query": {
    "expression": "#name = :name",
    "expressionNames": {
      "#name": "name",
    },
    "expressionValues": {
      ":name": {
        "S": "test",
      },
    },
  },
  "select": "ALL_ATTRIBUTES",
}
`;

exports[`dynamodb resolvers something 2`] = `
[
  {
    "a": 10,
  },
]
`;
